docs/design/kubeadm-api.md

```diff
@@ -0,0 +1,275 @@
+## Kubeadm Phases and API
+
+### Refactoring kubeadm
+We all want kubeadm modular and componentized, here’s the proposal for that!
+I’m not writing down all reasons for doing this; they are already available in the blog post: https://docs.google.com/document/d/1bpEGhdg8t-2Ae-IL_ezZIPklMHpZLWwnoAPMgLwT3oU
+
+### Phases
+We want to identify the phases every single Kubernetes deployment goes through in order to get the cluster up and running.
+
+In this process, I've been thinking a lot about security and tried to put that first. Also, I've read the 
+kops, kargo and kube-up code in detail to see how they are doing these things. We don't want to make an API that's incompatible with those tools.
+
+So far (for kubeadm) I've identified these:
+
+#### Certificates
```

Comment:
Where is the code for this? Is using a separate CA required?
I'd very much like to have more information about how you have it setup on GCE regarding that.